src/util/castJsonTypes.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Values in the JSON should be appropriately typed however this may not always be the case.
+ * For instance, certain values of a continuous trait may be strings, and we should cast
+ * these to numbers. See https://github.com/nextstrain/auspice/issues/1626 for more discussion
+ * of this particular case. Feel free to add more checks / casts to this function!
+ * @param {Object} metadata
+ * @param {Object} tree
+ * @returns {undefined} Any type casting is in-place
+ */
+export const castIncorrectTypes = (metadata, tree) => {
+  try {
+    const continuousKeys = new Set();
+    Object.entries(metadata.colorings || {}).forEach(([key, details]) => {
+      if (details.type==="continuous") {
+        continuousKeys.add(key);
+      }
+    });
+    tree.nodes.forEach((node) => {
+      Object.entries(node.node_attrs || {}).forEach(([key, details]) => {
+        if (continuousKeys.has(key)) {
+          if ((typeof details.value) !== "number") {
+            if (details.value==="" || isNaN(details.value)) { // Note: Number("")=0
```

Comment:
Any string with only whitespaces also get converted to 0 because Number() ignores leading and trailing whitespace/line terminators. 

`details.value.trim() === ""` should catch those cases here.

Reply:
Should we also check for `null` and boolean values here? 
`null` gets converted to 0 and booleans get converted to 1/0.

Reply:
Good catches! I'll refactor next week to make any values other than strings & numbers `undefined`, and also trim the strings & deal with infinities.

Reply:
As of [c7ac36f](https://github.com/nextstrain/auspice/pull/1655/commits/c7ac36f12ff9a282964831ec32df77b44ce29bef) those examples will now all be `undefined`.